### PR TITLE
[FS] Add MaxTipAge to console and improve IBD reporting

### PR DESCRIPTION
**Pull request by quantumagi**
See https://dev.azure.com/Stratisplatformuk/StratisBitcoinFullNode/_workitems/edit/4076.

image

### CODE REVIEW

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 1]
```diff
@@ -1369,7 +1373,12 @@ private void AddComponentStats(StringBuilder log)
 
             lock (this.peerLock)
             {
-                if (this.isIbd) log.AppendLine("IBD Stage");
+                long currentTime = DateTimeProvider.Default.GetTime();
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: This should use the an injected provider

**Comment by dangershony** [comment]
What is the purpose of knowing the tip age?

**Comment by quantumagi** [comment]
> What is the purpose of knowing the tip age?

Having the Tip Age and Max Tip Age displayed together with the IBD stage serves as a reminder to the user how this affects being in IBD.

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: This mimics `InitialBlockDownloadState` which used the default provider for the same calculation:

image

**Comment by dangershony** [comment]
@quantumagi ok I am convinced its a good idea... just one more comment in the code

**Comment by dangershony** [comment]
Perhaps it was wrong before no?
There is another change coming that inject the DTP to consensus so it will be available anyway.

**Pull request merged**